### PR TITLE
Restore on new cluster

### PR DESCRIPTION
**Pull request by dAdAbird**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | failed |
| mongodb 4.0 | failed |
| mongodb 4.2 | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | failed |
| mongodb 4.0 | failed |
| mongodb 4.2 | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | failed |
| mongodb 4.0 | failed |
| mongodb 4.2 | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | failed |
| mongodb 4.0 | failed |
| mongodb 4.2 | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | failed |
| mongodb 4.0 | failed |
| mongodb 4.2 | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | failed |
| mongodb 4.0 | failed |
| mongodb 4.2 | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | passed |
| mongodb 4.0 | failed |
| mongodb 4.2 | failed |

File: e2e-tests/functions [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    local cmd=$1
+    local rs=$2
+
+    docker-compose -f $COMPOSE_PATH exec -T "${rs}01" mongo "mongodb://${BACKUP_USER}:${MONGO_PASS}@localhost/?replicaSet=${rs}" --quiet --eval="${cmd}"
+}
+
+wait_backup() {
+    set +o xtrace
+
+    BCP_NAME=$(mongo_run "db.getSiblingDB(\"admin\").pbmBackups.find({}, {name: 1, _id: 0}).sort({start_ts: -1}).limit(1)" "cfg" | tail -n 1 | awk -F\" '{print$4}')
```
[the remark this answers is not in the document]
Comment (dAdAbird) [reply e2e-tests/functions]: Along with the requested data mongo shell spits out quite a bid of some rubbish. So, yes, it's needed.

File: e2e-tests/run-consistency [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+desc 'Start cluster'
+start_cluster "$MONGO_VERSION"
+
+docker-compose -f $COMPOSE_PATH start tests
+docker-compose -f $COMPOSE_PATH logs -f --no-color tests
+EXIT_CODE=$(docker-compose -f $COMPOSE_PATH ps -q tests | xargs docker inspect -f '{{ .State.ExitCode }}')
+
+destroy_cluster
+
+exit $EXIT_CODE
```
[the remark this answers is not in the document]
Comment (dAdAbird) [reply e2e-tests/run-consistency]: https://github.com/percona/percona-backup-mongodb/pull/348/commits/bac2b9a6479623114ba4011a6126c1e873cf32c8

File: e2e-tests/functions [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            agent-cfg01 agent-cfg02 agent-cfg03  agent-rs101 agent-rs102 agent-rs103 agent-rs201 agent-rs202 agent-rs203
+    docker-compose -f $COMPOSE_PATH up -d --build \
+            agent-cfg01 agent-cfg02 agent-cfg03  agent-rs101 agent-rs102 agent-rs103 agent-rs201 agent-rs202 agent-rs203
+
+}
+
+destroy_cluster() {
+    docker-compose -f $COMPOSE_PATH ps
+    docker-compose -f $COMPOSE_PATH down
+}
```
[the remark this answers is not in the document]
Comment (dAdAbird) [reply e2e-tests/functions]: https://github.com/percona/percona-backup-mongodb/pull/348/commits/bac2b9a6479623114ba4011a6126c1e873cf32c8

File: e2e-tests/functions [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+            mongo_run "db.getSiblingDB(\"admin\").pbmLock.find()" "cfg"
+            exit 1
+        fi
+    done
+    set -o xtrace
+}
+
+compare() {
+    if [ $1 != $2 ]; then
+        echo "$1 doesn't march $2"
```
[the remark this answers is not in the document]
Comment (dAdAbird) [reply e2e-tests/functions]: https://github.com/percona/percona-backup-mongodb/pull/348/commits/bac2b9a6479623114ba4011a6126c1e873cf32c8

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | passed |
| mongodb 4.0 | failed |
| mongodb 4.2 | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | passed |
| mongodb 4.0 | failed |
| mongodb 4.2 | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | passed |
| mongodb 4.0 | passed |
| mongodb 4.2 | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | failed |
| mongodb 4.0 | passed |
| mongodb 4.2 | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | passed |
| mongodb 4.0 | passed |
| mongodb 4.2 | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 4.0 | passed |
| mongodb 3.6 | passed |
| mongodb 4.2 | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | passed |
| mongodb 4.0 | passed |
| mongodb 4.2 | passed |

**Pull request merged**